### PR TITLE
Auto Update

### PR DESCRIPTION
**Pull request by DanielSvoboda**
# Update the emulator directly from the application.

In the settings menu, you can enable the "Check for Updates at Startup" option and choose the download channel:

- Stable: stable versions released approximately once a month (These versions are more reliable and tested)
- Unstable: pre-releases with updates almost daily (These may contain bugs and are less stable)

*not tested on macOS

# Images:

image

You can see the changelog by clicking the button. It is not available if you are using an unstable version and try to update to stable, as there will be no changelog for older versions.
image

The 'About' tab is now called 'Help' and you can also check for updates using this button.
image

Yiga also contributed the icon for the download and dump:
image

Settings: You can enable the "Check for Updates at Startup" option and choose the download channel Stable, Unstable
image

The translation files "tr_TR.ts" and "ar.ts" were also fixed.
It was inspired by the operation of rpcs3, none of its code was used, but I mention it in recognition.

# Technical information- Create autoUpdate in your fork
For the pre-release to work on a fork, you need to create a token, and change some links, here is the step by step:

https://github.com/settings/tokens/new
Note: SHADPS4_TOKEN_REPO
Expiration: No expiration
Select scopes: repo
Create and copy the code (it will only appear once, so it is a good idea to put it in a notepad until you use it)
[do not send it to anyone]

the link below change 'shadPS4-emu' to your user
https://github.com/shadPS4-emu/shadPS4/settings/secrets/actions
New repository secret
name: SHADPS4_TOKEN_REPO
Secret: paste the code you copied in the previous step

and update the 3 links in check_update.cpp to your repository
change repository name in build.yml
`body: "Full Changelog: ... https://github.com/shadps4-emu/shadPS4/compare ...`
You may also need to change the branch name in:
`if: github.ref == 'refs/heads/main' && github.event_name == 'push'`

```
on:
  push:
    branches: [ "main" ]
```
----

Special thanks: LeDragoX | Vinicius Rangel | Yiga

### CODE REVIEW

**Comment by rambass658** [comment]
Dont work. Update to previous ver.

File: src/qt_gui/checkUpdate.cpp
Comment (polybiusproxy) [inline src/qt_gui/checkUpdate.cpp:1]: Filename should be `check_update.cpp`/`check_update.h`

File: src/common/scm_rev.cpp.in [hunk 1]
```diff
@@ -6,12 +6,14 @@
 #define GIT_REV      "@GIT_REV@"
 #define GIT_BRANCH   "@GIT_BRANCH@"
 #define GIT_DESC     "@GIT_DESC@"
+#define BUILD_DATE "@BUILD_DATE@"
```
Comment (polybiusproxy) [inline src/common/scm_rev.cpp.in]: Missing indentation

**Review by polybiusproxy — CHANGES_REQUESTED** [review]

**Comment by polybiusproxy** [comment]
You haven't updated the filenames on the Qt files

**Comment by DanielSvoboda** [comment]
> You haven't updated the filenames on the Qt files

Which file is missing?
I renamed the 2 you mentioned check_update.cpp/check_update.h
image

**Pull request merged**